compiler/rustc_lint/src/levels.rs

```diff
@@ -259,6 +259,14 @@ impl<'s> LintLevelsBuilder<'s> {
         let sess = self.sess;
         let bad_attr = |span| struct_span_err!(sess, span, E0452, "malformed lint attribute input");
         for (attr_index, attr) in attrs.iter().enumerate() {
+            if attr.has_name(sym::automatically_derived) {
+                self.current_specs_mut().insert(
+                    LintId::of(SINGLE_USE_LIFETIMES),
+                    (Level::Allow, LintLevelSource::Default),
+                );
+                continue;
+            }
```

Comment:
In that case, the macro code uses the lifetime definitions span for all uses.  This prevents using `in_external_macro`.  Fixing that will require fixing the macro.

Reply:
Ok, I don't want to dig into this too much right now, but eventually this will need to be changed to use something other than `automatically_derived` because non-built-in macros don't typically use `automatically_derived`.